mm2src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -0,0 +1,734 @@
+    let ordermatch_ctx = OrdermatchContext::from_ctx(ctx).unwrap();
+    let maker_orders_guard = ordermatch_ctx.my_maker_orders.lock().await;
+    let maker_orders = maker_orders_guard.clone();
+    drop(maker_orders_guard);
+
+    info!("nb_orders: {}", maker_orders.len());
+
+    let mut futures_order_update = Vec::with_capacity(0);
+    // Iterating over maker orders and update order that are present in cfg as the key_trade_pair e.g KMD/LTC
+    for (uuid, value) in maker_orders.into_iter() {
+        let key_trade_pair = TradingPair::new(value.base.clone(), value.rel.clone());
+        match cfg.get(&key_trade_pair.as_combination()) {
+            Some(coin_cfg) => {
+                let (result_sender, result_receiver) = futures::channel::oneshot::channel();
+                let cloned_infos = (
+                    ctx.clone(),
+                    rates_registry.get_cex_rates(coin_cfg.base.clone(), coin_cfg.rel.clone()),
+                    key_trade_pair.clone(),
+                    coin_cfg.clone(),
+                );
+                spawn(execute_update_order(uuid, value, result_sender, cloned_infos));
+                futures_order_update.push(result_receiver);
+                memoization_pair_registry.insert(key_trade_pair.as_combination());
+            },
+            _ => continue,
+        }
+    }
+
+    let all_updated_orders_tasks = futures::future::join_all(futures_order_update.into_iter());
+    let _results_order_updates = all_updated_orders_tasks.await;
+
+    let mut futures_order_creation = Vec::with_capacity(0);
+    // Now iterate over the registry and for every pairs that are not hit let's create an order
+    for (trading_pair, cur_cfg) in cfg.into_iter() {
+        match memoization_pair_registry.get(&trading_pair) {
+            Some(_) => continue,
+            None => {
+                let (result_sender, result_receiver) = futures::channel::oneshot::channel();
+                let ctx_cloned = ctx.clone();
+                let rates_infos = rates_registry.get_cex_rates(cur_cfg.base.clone(), cur_cfg.rel.clone());
+                spawn(async move {
+                    let res = match create_single_order(rates_infos, cur_cfg, trading_pair.clone(), ctx_cloned).await {
+                        Ok(resp) => resp,
+                        Err(err) => {
+                            error!("{} order cannot be created - {}", trading_pair, err);
+                            false
+                        },
+                    };
+                    match result_sender.send(res) {
+                        Ok(_) => {},
+                        Err(_) => error!("{} order cannot be created", trading_pair),
+                    };
+                });
+                futures_order_creation.push(result_receiver);
+            },
+        };
+    }
+    let all_created_orders_tasks = futures::future::join_all(futures_order_creation.into_iter());
+    let _results_order_creations = all_created_orders_tasks.await;
```

Comment:
For the moment, it's unused, but I would love to add a telegram bot that would use the future result to send notifications to the user.